src/lib/storage/vector_compression/simd_bp128/simd_bp128_packing.hpp

```diff
@@ -21,8 +21,8 @@ class SimdBp128Packing {
   static constexpr auto meta_block_size = block_size * blocks_in_meta_block;
 
  public:
-  static void write_meta_info(const uint8_t* in, uint128_t* out);
-  static void read_meta_info(const uint128_t* in, uint8_t* out);
+  static void write_meta_info(const uint8_t* in, uint128_t* const out);
```

Comment:
Isn't in double const and the output just const on the pointer?

Reply:
Yepp

Reply:
Ok, the `in` makes sense to me now. What does `uint128_t* const out` actually mean? Can the pointer we non-const but the content is supposed to be const? Doesn't make sense to me.

Reply:
> Can the pointer we non-const but the content is supposed to be const?

Yes. In that case, you could make the pointer point somewhere else, but you couldn't change the value that is pointed to.

However, in the code above, we have the exact opposite: The pointer cannot be changed, but the value that is pointed to can (const applies to left)

Reply:
So the `const auto* const separator` we have all over the place is an example where the const applies to the right side (for the left const it's clear as there is not left of it, but the the right side that's "transitive"???).
But then again here in the SIMD code, it applies to the pointer.

Hmpf ...

Reply:
It's a const pointer to a const thing, i.e., neither can the pointer be changed to point somewhere else nor can the "thing" be changed itself.

What do you mean with "transitive"?

Did you watch the east const video? I think that clarifies it.

Reply:
I've seen it. But I am not sure I fully understood it. :)

What bothers me, is that we have two syntaxes in the same line (if I get it right): `const uint8_t* const in, uint128_t* const out` ... for in, the const is on the left side, for out it's on the right side. West coast, east coast. Or am I again mistaken?

Reply:
That's why west const sucks. ;)

If we were using east const only, it would look like this:

`uint8_t const *const in, uint128_t *const out`

Looks like the [clang-format folks are working on it](https://reviews.llvm.org/D69764). It would be interesting to see how many lines would have to change.

Reply:
Definitely something I have to get used to. If clang-format will solve that for us: fine. But until then we should make sure that at least we don't mix both styles in a single line.